packages/snaps-execution-environments/src/common/BaseSnapExecutor.ts

```diff
@@ -369,12 +369,12 @@ export class BaseSnapExecutor {
       });
     };
 
-    const provider = new StreamProvider(this.rpcStream, {
+    const provider = new SnapProvider(this.rpcStream, {
       jsonRpcStreamName: 'metamask-provider',
       rpcMiddleware: [createIdRemapMiddleware()],
     });
 
-    await provider.initialize();
+    provider.initializeSync();
```

Comment:
This is the main change of this PR alongside the introduction of the `SnapProvider` wrapper class